tst/EnergyPlus/unit/EconomicLifeCycleCost.unit.cc

```diff
@@ -415,3 +421,172 @@ TEST_F(EnergyPlusFixture, EconomicLifeCycleCost_ProcessMaxInput)
     EXPECT_EQ(1.099, UseAdjustment(1).Adjustment(99));
     EXPECT_EQ(1.100, UseAdjustment(1).Adjustment(100));
 }
+
+TEST_F(EnergyPlusFixture, EconomicLifeCycleCost_ComputeEscalatedEnergyCosts)
+{
+    lengthStudyYears = 5;
+
+    numCashFlow = 1;
+    CashFlow.allocate(numCashFlow);
+    CashFlow(1).pvKind = pvkEnergy;
+    CashFlow(1).Resource = 1001;
+    CashFlow(1).yrAmount.allocate(lengthStudyYears);
+    CashFlow(1).yrAmount(1) = 100;
+    CashFlow(1).yrAmount(2) = 100;
+    CashFlow(1).yrAmount(3) = 100;
+    CashFlow(1).yrAmount(4) = 100;
+    CashFlow(1).yrAmount(5) = 100;
+
+    numResourcesUsed = 1;
+
+    EscalatedEnergy.allocate(lengthStudyYears, 1);
+    EscalatedTotEnergy.allocate(lengthStudyYears);
+    EscalatedTotEnergy = 0.0;
+
+    ComputeEscalatedEnergyCosts();
+    EXPECT_NEAR(EscalatedEnergy(1, 1), 100., 0.001);
+    EXPECT_NEAR(EscalatedEnergy(2, 1), 100., 0.001);
+    EXPECT_NEAR(EscalatedEnergy(3, 1), 100., 0.001);
+    EXPECT_NEAR(EscalatedEnergy(4, 1), 100., 0.001);
+    EXPECT_NEAR(EscalatedEnergy(5, 1), 100., 0.001);
```

Comment:
Probably not a good choice for a unit test (using all 100s) where you don't know if the correct CashFlow(1).yrAmount(x) was used. Would rather see 100, 200, 300... or 101, 102, 103... Might not be worth the effort.

Reply:
Good idea.